shared/menubar/space-warning.js

```diff
@@ -1,26 +1,59 @@
 // @flow
 import * as React from 'react'
 import * as Kb from '../common-adapters'
+import * as FsTypes from '../constants/types/fs'
 
-export type BannerType = 'warning' | 'failure' | 'none'
 type Props = {
-  onClose: () => void,
   onRetry: () => void,
-  bannerType: BannerType,
+  diskSpaceStatus: FsTypes.DiskSpaceStatus,
 }
+type State = {
+  hidden: boolean,
+}
+
+class SpaceWarning extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {hidden: false}
+  }
 
-const Banner = (props: Props) =>
-  props.bannerType !== 'none' && (
-    <Kb.Banner
-      onClose={props.onClose}
-      text={
-        props.bannerType === 'warning'
-          ? 'You have less than 1 GB of' + ' storage space. Make some space, or unsync some folders.'
-          : 'You are' + ' out of storage space. Unsync some folders, or make some space then'
-      }
-      color={props.bannerType === 'warning' ? 'blue' : 'red'}
-      actions={[...(props.onRetry ? [{onClick: props.onRetry, title: 'retry' + ' the sync.'}] : [])]}
-    />
-  )
+  _close = () => {
+    this.setState(() => ({
+      hidden: true,
+    }))
```

Comment:
The state will be gone once the component is unmounted (which I think happens when the widget hides). Did you mean to have it come back next time the widget is open?

Reply:
I did :/ - any suggestions for that?

Reply:
Oh I think I might have misunderstood. It sounds like the current behavior is it doesn't come back when the widget is closed and re-opened, and you want that to happen?

Reply:
I did mean for the state to come back so that the banner does not come back.

Reply:
Oh I see. I think instead of using component state, we'd just need to move it into the redux store. The SFMI banner uses a similar mechanism.